app/controllers/manifest_uploads_controller.rb

```diff
@@ -1,5 +1,46 @@
 class ManifestUploadsController < ApplicationController
   def new
+  end
+
+  def create
+    manifest = find_or_initialize_manifest
+
+    if manifest.save
+      flash[:notice] = "Upload for manifest #{manifest.tracking_number} submitted successfully."
+      redirect_to root_path
+    end
+  end
+
+  private
+
+  def find_or_initialize_manifest
+    if params[:manifest_id]
```

Comment:
this controller is web-only. If someone were to take this action via API, it would be in the context of a POST `/manifest` or PATCH `/manifests` action (either created a new manifest with an upload or updating an existing one)

So, there isn't anything to be consistent with exactly. But, if you think something should be changed here for clarity etc, let me know!